src/Service/S3/src/Signer/SignerV4ForS3.php

```diff
@@ -86,8 +95,10 @@ protected function convertBodyToStream(SigningContext $context): void
         }
 
         // no need to stream small body. It's simple to convert it to string directly
-        if ($contentLength < self::CHUNK_SIZE) {
-            $request->setBody(StringStream::create($body));
+        if ($contentLength < self::CHUNK_SIZE || !$this->sendChunkedBody) {
+            if ($body instanceof ReadOnceResultStream) {
+                $request->setBody(RewindableStream::create($body));
+            }
```

Comment:
What if the body is not an instance of ReadOnceResultStream?

Reply:
`$body` is initialized from `$body = $request->getBody()`.
If it's not an instance of `ReadOnceResultStream`, then we are good, we can rewind the stream.

Reply:
I fail to understand why though.. Before, we always converted to a `StringStream` if the body was small, but now we dont anymore. Was it a misstake before?

Reply:
Yes, I believe the conversation were not needed.

If I remember well, when I implemented the conversion to string, the `ReadOnceResultStream` did not exists and stream have to be converted into string at some point.

Now, thanks to the RewindableStream, the signature is able to works with all cases, but I forgot to revert upgrade this statement.